### PR TITLE
feat(split_functional_groups): Adds fxn

### PR DESCRIPTION
**Pull request by kellijohnson-NOAA**
# What is the feature?
* The added function will split the functional_group information into a data frame with multiple columns. 

# How have you implemented the solution?
* The added function will split the functional_group information into a data frame with multiple columns. The function is designed to be used with the data to add columns but is exported as its own function so that users can check it is working as intended.

# Does the PR impact any other area of the project, maybe another repo?
* @Bai-Li-NOAA I am not sure when/where you want to integrate this into the output, which is why this is just a draft PR. Feel free to comment and let me know what I can do to finish it.

### CODE REVIEW

**Review by Bai-Li-NOAA — COMMENTED** [review]
Thanks, @kellijohnson-NOAA. This looks great. I only have a few minor suggestions related to the documentation.

As for where to integrate this into the output, I'm thinking we could call it within `get_functional_groups()`. Instead of passing a vector of functional group names to `load_model()`, we’d now pass a tibble with three columns (i.e., functional_group, species, and group)

Feel free to work on this during the next co-working session!

File: R/split_functional_groups.R [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+#' )
+#' split_functional_groups(c("spiny dogfish", "spiny dogfish juvenile"))
+#' data("ewe_nwatlantic_base", package = "ecosystemdata")
+#' split_functional_groups(unique(ewe_nwatlantic_base[["functional_group"]]))
+split_functional_groups <- function(x) {
+  if (!inherits(x, what = "character")) {
+    cli::cli_abort("{.var x} must be a character vector not a {class(x)}.")
+  }
+  groups <- unique(x) |>
+    gsub(pattern = "([a-z])([[:upper:]])", replacement = "\\1 \\2") |>
```
Comment (Bai-Li-NOAA) [inline R/split_functional_groups.R:29]: Could you add a brief, plain-language comment explaining lines 29-31, similar to the one for lines 33-34?

File: tests/testthat/test-split_functional_groups.R [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+  expect_equal(
+    object = split_functional_groups(c("spiny dogfish0", "spiny dogfish 2")),
+    expected = tibble::tibble(
+      functional_group = c("spiny dogfish0", "spiny dogfish 2"),
+      species = c("spiny dogfish", "spiny dogfish"),
+      group = c("0", "2")
+    )
+  )
+
+  #' @description Test that split_functional_groups(x) returns y.
```
Comment (Bai-Li-NOAA) [inline tests/testthat/test-split_functional_groups.R:25]: Could you provide additional details in the description for this test?

File: tests/testthat/test-split_functional_groups.R [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+test_that("split_functional_groups() returns correct outputs for edge cases", {
+  #' @description Test that split_functional_groups(x) uses sentence case rather
+  #' than camel case.
+  input <- c("SpinyDogfish0", "SpinyDogfish 2_3")
+  test <- split_functional_groups(input)
+  expect_equal(
+    object = test[["species"]],
+    expected = c("Spiny Dogfish", "Spiny Dogfish")
+  )
+  expect_equal(
```
Comment (Bai-Li-NOAA) [inline tests/testthat/test-split_functional_groups.R:61]: Could you add a one-sentence description for the test?

File: R/split_functional_groups.R (hunk 1 above)
Comment (kellijohnson-NOAA) [reply R/split_functional_groups.R:29]: Thank you for the suggestion. Comments are now present.

File: tests/testthat/test-split_functional_groups.R (hunk 2 above)
Comment (kellijohnson-NOAA) [reply tests/testthat/test-split_functional_groups.R:25]: done

File: tests/testthat/test-split_functional_groups.R (hunk 3 above)
Comment (kellijohnson-NOAA) [reply tests/testthat/test-split_functional_groups.R:61]: done

**Comment by kellijohnson-NOAA** [comment]
The functional groups are now appropriately ordered according to how the input vector was ordered, I had it wrong before. Also, the species and groups columns are now returned when loading the data. The only thing left to do is update the data frames included in the package.
